### PR TITLE
Added single-satellite simulation to usb_console

### PR DESCRIPTION
**Pull request by tanishqaggarwal**
Be sure to check out configs to see the config changes.

Also, some bad news: currently everything seems to be running at ~0.6x real time, despite no state exchange happening between MATLAB and the Teensies. I'll investigate this when I find the time

### CODE REVIEW

File: python/usb_console/cmdprompt.py [hunk 1]
```diff
@@ -24,10 +23,12 @@ def __init__(self, devices, radios, sim, exit_fn):
 
         # By default, if it's available, set the prompt to be commanding the Flight Controller.
         try:
-            self.cmded_device = self.devices['FlightControllerLeader']
-
+            self.cmded_device = self.devices['FlightController']
         except KeyError:
-            self.cmded_device = random.choice(list(self.devices.values()))
```
Comment (tanishqaggarwal) [inline python/usb_console/cmdprompt.py:30]: randomly choosing a device was so stupid

File: python/usb_console/run_simulation.py [hunk 2]
```diff
@@ -117,7 +118,7 @@ def set_up_radios(self):
                 radio_data_name = radio_connected_device + "_radio"
                 radio_datastore = Datastore(radio_data_name, self.simulation_run_dir)
                 radio_logger = Logger(radio_data_name, self.simulation_run_dir)
-                radio_session = RadioSession(radio_connected_device, radio_datastore, radio_logger, self.radio_keys_config)
+                radio_session = RadioSession(radio_name, radio_datastore, radio_logger, self.radio_keys_config)
```
Comment (tanishqaggarwal) [inline python/usb_console/run_simulation.py:121]: This is to fix name clashes between the "FlightController" Teensy and the "FlightController" radio when commanding state via the command prompt

File: python/usb_console/simulation.py [hunk 3]
```diff
@@ -136,3 +138,81 @@ def stop(self, data_dir):
 
         with open(data_dir + "/simulation_log.txt", "w") as fp:
             fp.write(self.log)
+
+class SingleSimulation(Simulation):
```
Comment (tanishqaggarwal) [inline python/usb_console/simulation.py]: @kkrol27 this kind of abstraction is how the company we love built our most favorite, gnarly test product

**Comment by tanishqaggarwal** [comment]
@nhz2 As Kyle predicted, this solution ended up being pretty hacky. Would it be a good idea for MATLAB to more natively support a single-sat simulation?

**Comment by nhz2** [comment]
Can't you just sim the leader FC using `update_FC_state` in matlab and only have the follower in HITL/SITL?

File: python/usb_console/simulation.py (hunk 3 above)
Comment (kylekrol) [reply python/usb_console/simulation.py]: Yep looks like it haha

**Comment by kylekrol** [comment]
> Can't you just sim the leader FC using `update_FC_state` in matlab and only have the follower in HITL/SITL?

Just to emphasize this, the way I sorta had seen this going down was, as Nathan suggested, have the MATLAB `update_FC_state` act as a minimum working example of a FC stepping through a completely nominal mission.

That being said, it may be advantageous to take a single sat sim even further. In such a case we could have the second spacecraft simple be a dummy target. It would always have it's docking end pointed directly at the follower spacecraft and we wouldn't need to sim any FC behavior, actuators, sensors, or physics (except for a point mass orbit propagator of course). I'm not sure how possible this is now, but I could take some time to defamiliarize myself with the MATLAB code we have now to see if this is possible.

Let me know if anyone has additional thoughts, comments, concerns.

**Comment by tanishqaggarwal** [comment]
I think that's exactly what's going on right now @kkrol27

**Review by kylekrol — APPROVED** [review]

**Pull request merged**